plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtScaleVmCommandWrapper.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2021 The Apache Software Foundation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cloud.hypervisor.kvm.resource.wrapper;
+
+import com.cloud.agent.api.Answer;
+import com.cloud.agent.api.ScaleVmAnswer;
+import com.cloud.agent.api.ScaleVmCommand;
+import com.cloud.agent.api.to.VirtualMachineTO;
+import com.cloud.hypervisor.kvm.resource.LibvirtComputingResource;
+import com.cloud.hypervisor.kvm.resource.LibvirtVmMemoryDeviceDef;
+import com.cloud.resource.CommandWrapper;
+import com.cloud.resource.ResourceWrapper;
+import org.apache.cloudstack.utils.bytescale.ByteScaleUtils;
+import org.libvirt.Connect;
+import org.libvirt.Domain;
+import org.libvirt.LibvirtException;
+
+@ResourceWrapper(handles = ScaleVmCommand.class)
+public final class LibvirtScaleVmCommandWrapper extends CommandWrapper<ScaleVmCommand, Answer, LibvirtComputingResource> {
+
+    @Override
+    public Answer execute(ScaleVmCommand command, LibvirtComputingResource libvirtComputingResource) {
+        VirtualMachineTO vmSpec = command.getVirtualMachine();
+        String vmName = vmSpec.getName();
+        Connect conn = null;
+
+        long newMemory = ByteScaleUtils.bytesToKib(vmSpec.getMaxRam());
+        int newVcpus = vmSpec.getCpus();
+        String vmDefinition = vmSpec.toString();
+        String scalingDetails = String.format("%s memory to [%s KiB] and cpu cores to [%s]", vmDefinition, newMemory, newVcpus);
+
+        try {
+            LibvirtUtilitiesHelper libvirtUtilitiesHelper = libvirtComputingResource.getLibvirtUtilitiesHelper();
+
+            conn = libvirtUtilitiesHelper.getConnectionByVmName(vmName);
+            Domain dm = conn.domainLookupByName(vmName);
+
+            long currentMemory = LibvirtComputingResource.getDomainMemory(dm);
+            long runningVcpus = libvirtComputingResource.countDomainRunningVcpus(dm);
+            long memoryToAttach = newMemory - currentMemory;
+
+            logger.debug(String.format("Scaling %s.", scalingDetails));
+
+            if (memoryToAttach > 0) {
+                String memoryDevice = new LibvirtVmMemoryDeviceDef(newMemory - LibvirtComputingResource.getDomainMemory(dm)).toString();
```

Comment:
```suggestion
                String memoryDevice = new LibvirtVmMemoryDeviceDef(memoryToAttach).toString();
```
I think that there is no need to retrieve the domain's memory again

Reply:
@slavkap done, thanks for the review.